### PR TITLE
Added Richardson-Lucy deconvolution

### PR DESCRIPTION
**Pull request by jakubwro**
Hi, please review. If you think it's worth merging I'll update docs.

### CODE REVIEW

File: doc/lucy-cameraman.jl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+using Images, TestImages, Deconvolution, Random, FFTW
+
+Random.seed!(42)
+
+img = channelview(testimage("cameraman"))
+
+(N, M) = size(img)
+@assert N == M
+@assert floor(log2(N)) == ceil(log2(N)) # ensure size is power of 2
```
Comment (giordano) [inline doc/lucy-cameraman.jl]: There is a function in Base to check this:
```suggestion
@assert ispow2(N) # ensure size is power of 2
```

**Comment by giordano** [comment]
Wow, this is great, thank you very much!  I'll review later.

This PR also reminds me that I should move documentation to Documenter :sweat:

**Review by giordano — APPROVED** [review]
I cannot comment on the algorithm in particular as I'm not familiar with it, but the code looks good, thank you so much!

> If you think it's worth merging I'll update docs.

Yes, please :slightly_smiling_face:

File: doc/lucy-cameraman.jl (hunk 1 above)
Comment (jakubwro) [reply doc/lucy-cameraman.jl]: Great, I will update docs before December, I’m traveling now. Maybe someone else will review the algorithm till that moment.

**Comment by jakubwro** [comment]
Hi @giordano , I added docs and more illustrative usage example

**Comment by giordano** [comment]
Awesome, I'll have a look in the next few days!

**Comment by jakubwro** [comment]
@giordano, I readded docs, but I have a feeling that messed something with git  #history. Please have a look.

**Comment by giordano** [comment]
The history is not a problem, we can just squash everything :wink:

I followed your struggle, I'm really sorry for this.  I can feel your pain, I had similar problems a few days ago.  In theses cases I think that it's easier to do a clean merge of the target branch on yours rather than rebasing.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,4 @@
 *.jl.cov
 *.jl.*.cov
 *.jl.mem
-/doc/_build/
+/doc/build/
```
Comment (jakubwro) [inline .gitignore:4]: @abhishalya , Documenter uses different output folder and .gitignore should be changed. Sorry for not catching that during review.

**Comment by jakubwro** [comment]
> The history is not a problem, we can just squash everything wink
> 
> I followed your struggle, I'm really sorry for this. I can feel your pain, I had similar problems a few days ago. In theses cases I think that it's easier to do a clean merge of the target branch on yours rather than rebasing.

Right, it's not the first time I have this problem with rebase.

File: .gitignore (hunk 2 above)
Comment (abhishalya) [reply .gitignore:4]: Ah, right. Sorry I missed that too. Thanks for correcting that :)

**Comment by giordano** [comment]
Documentation is available at https://juliadsp.github.io/Deconvolution.jl/dev/.  Thank you so much @jakubwro and @abhishalya!  This new method looks very good!

**Pull request merged**